endpointmanager/healthitproduct.go

```diff
@@ -0,0 +1,312 @@
+	APISyntax             string    // the format of the information provided by the API, for example, REST, FHIR STU3, etc.
+	APIURL                string    // the URL to the API documentation for the product.
+	CertificationCriteria []string  // the ONC criteria that the product was certified to, for example, ["170.315 (g)(7)", "170.315 (g)(8)", "170.315 (g)(9)"]
+	CertificationStatus   string    // the ONC certification status, for example, "Active", "Retired", "Suspended by ONC", etc.
+	CertificationDate     time.Time
+	CertificationEdition  string // the product's certification edition for the ONC Health IT certification program, for example, "2014", "2015".
+	LastModifiedInCHPL    time.Time
+	CHPLID                string // the product's unique ID within the CHPL system.
+	CreatedAt             time.Time
+	UpdatedAt             time.Time
+}
+
+// GetHealthITProduct gets a HealthITProduct from the database using the database id as a key.
+func GetHealthITProduct(id int) (*HealthITProduct, error) {
+	var hitp HealthITProduct
+	var locationJSON []byte
+	var certificationCriteriaJSON []byte
+
+	sqlStatement := `SELECT id,
+							name,
+							version,
+							developer,
+							location,
+							authorization_standard,
+							api_syntax,
+							api_url,
+							certification_criteria,
+							certification_status,
+							certification_date,
+							certification_edition,
+							last_modified_in_chpl,
+							chpl_id,
+							created_at,
+							updated_at
+					FROM healthit_products WHERE id=$1`
+	row := db.QueryRow(sqlStatement, id)
+
+	err := row.Scan(
+		&hitp.id,
+		&hitp.Name,
+		&hitp.Version,
+		&hitp.Developer,
+		&locationJSON,
+		&hitp.AuthorizationStandard,
+		&hitp.APISyntax,
+		&hitp.APIURL,
+		&certificationCriteriaJSON,
+		&hitp.CertificationStatus,
+		&hitp.CertificationDate,
+		&hitp.CertificationEdition,
+		&hitp.LastModifiedInCHPL,
+		&hitp.CHPLID,
+		&hitp.CreatedAt,
+		&hitp.UpdatedAt)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(locationJSON, &hitp.Location)
+	if err != nil {
+		return nil, err
+	}
+
+	err = json.Unmarshal(certificationCriteriaJSON, &hitp.CertificationCriteria)
+
+	return &hitp, err
+}
+
+// GetHealthITProductUsingNameAndVersion gets a HealthITProduct from the database using the given url as a key.
+func GetHealthITProductUsingNameAndVersion(name string, version string) (*HealthITProduct, error) {
+	var hitp HealthITProduct
+	var locationJSON []byte
+	var certificationCriteriaJSON []byte
+
+	sqlStatement := `SELECT id,
+							name,
+							version,
+							developer,
+							location,
+							authorization_standard,
+							api_syntax,
+							api_url,
+							certification_criteria,
+							certification_status,
+							certification_date,
+							certification_edition,
+							last_modified_in_chpl,
+							chpl_id,
+							created_at,
+							updated_at
+					FROM healthit_products WHERE name=$1 AND version=$2`
```

Comment:
the database is defined so that the pair of the name and version are unique:

```
CREATE TABLE healthit_products (
    id                      SERIAL PRIMARY KEY,
    name                    VARCHAR(500),
    version                 VARCHAR(500),
    developer               VARCHAR(500),
    location                JSONB,
    authorization_standard  VARCHAR(500),
    api_syntax              VARCHAR(500),
    api_url                 VARCHAR(500),
    certification_criteria  JSONB,
    certification_status    VARCHAR(500),
    certification_date      DATE,
    certification_edition   VARCHAR(500),
    last_modified_in_chpl   DATE,
    chpl_id                 VARCHAR(500),
    created_at              TIMESTAMPTZ NOT NULL DEFAULT NOW(),
    updated_at              TIMESTAMPTZ NOT NULL DEFAULT NOW(),
    CONSTRAINT healthit_product_info UNIQUE(name, version)
);
```

-> `CONSTRAINT healthit_product_info UNIQUE(name, version)`

So structurally, yes. 

It *should* be a unique combination in real life too. I could add `developer` to that list as well if you think that's necessary.

Reply:
I think that is alright so long as there is nothing that we want to keep track of over time that doesn't change with the version. Eg: If for some reason we wanted to track something like api_url, which I can't imagine is concretely coupled to the name/version combo.